rfcs/react-components/convergence/provider-style-overrides.md

```diff
@@ -0,0 +1,132 @@
+# RFC: Provider style overrides
+
+@ling1726
+
+## Summary
+
+Proposes two methods of overriding the default styles on the `FluentProvider`
+
+## Problem statement
+
+The `FluentProvider` ships with [default styles](https://github.com/microsoft/fluentui/blob/656f002f3062c0a7962fb23eb165dec64ecc1509/packages/react-components/react-provider/src/components/FluentProvider/useFluentProviderStyles.ts#L11-L18) which can be too opinionated for some customers, especially those who have
+strict requirements on global styles in their application.
+
+Currently it's not possible to use `makeStyles` to override these default styles. See the example below:
+
+```tsx
+import { makeStyles } from '@fluentui/react-components';
+
+const useProviderOverrides = makeStyles({
+  provider: {
+    // ⚠️ requires marginRight RTL style
+    marginLeft: '1px',
+  },
+});
+
+const App = () => {
+  const styles = useProviderOverrides();
+  return <FluentProvider className={styles.provider}>{children}</FluentProvider>;
+};
+```
+
+The above example is problematic because the `TextDirectionProvider` is rendered
+internally inside the `FluentProvider`. The provider is required to let `makeStyles` know to resolve and apply RTL
+styles to DOM.
+
+While it is possible to override these styles using global CSS in a stylesheet by targeting the provider, that would
+not handle portals which use the same default styles as the closest `FluentProvider` ancestor.
+
+## Detailed Design or Proposal
+
+### Document usage of extra providers
+
+We can document how to override `FluentProvider` styles using the required extra providers so that the
+default provider styles can be overriden.
+
+```tsx
+import { makeStyles, RendererProvider, TextDirectionProvider } from '@fluentui/react-components';
+
+const useProviderOverrides = makeStyles({
+  provider: {
+    marginLeft: '1px',
+  },
+});
+
+const App = ({ dir }) => {
+  const styles = useProviderOverrides();
+  return (
+    <FluentProvider dir={dir} className={styles.provider}>
+      {children}
+    </FluentProvider>
+  );
+};
+
+const AppWrapper = () => {
+  return (
+    <TextDirectionProvider dir="rtl">
+      <App dir="rtl" />
+    </TextDirectionProvider>
+  );
+};
+```
+
+#### Pros and Cons
+
+- 👍 No extra code changes necessary
+- 👍 Users will understand what the problem is
+- 👍 API Consistency (The same pattern must be adopted if user wants their own renderer with RendererProvider)
+
+- 👎 Extra effort to override one single component
```

Comment:
Not sure if it's a real problem as it will be done in a limited amount of places

Reply:
I would say it's still extra effort because it's not obvious and even with documentation it's hard to know to specifically look for it either